File: app/src/main/java/com/soullistener/viewtransfer/model/ViewTreeBean.java
package com.soullistener.viewtransfer.model;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * @author kuan
 * Created on 2018/12/20.
 * @description
 */
public class ViewTreeBean implements Serializable {

    /**
     * 子view
     */
    private ArrayList<ViewTreeBean> viewTreeBeans;


    /**
     * 本view信息
     */
    private ViewSaveBean viewSaveBean;

    public ViewTreeBean() {
    }

    public ViewTreeBean(ArrayList<ViewTreeBean> viewTreeBeans, ViewSaveBean viewSaveBean) {
        this.viewTreeBeans = viewTreeBeans;
        this.viewSaveBean = viewSaveBean;
    }

    public ArrayList<ViewTreeBean> getViewTreeBeans() {
        return viewTreeBeans;
    }

    public ViewSaveBean getViewSaveBean() {
        return viewSaveBean;
    }

    public int getChildViewCount() {
        return viewTreeBeans != null ? viewTreeBeans.size() : 0;
    }

    public void setViewSaveBean(ViewSaveBean viewSaveBean) {
        this.viewSaveBean = viewSaveBean;
    }

    public void setViewTreeBeans(ArrayList<ViewTreeBean> viewTreeBeans) {
        this.viewTreeBeans = viewTreeBeans;
    }

    public void printViewTreeBean() {

    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("ViewTreeBean ChildViewCount:"+getChildViewCount()+",");
        stringBuilder.append(viewSaveBean.toString());
        if (viewTreeBeans != null && viewTreeBeans.size() > 0) {
            for (ViewTreeBean viewTreeBean : viewTreeBeans) {
                stringBuilder.append(viewTreeBean.toString());
            }
        }

        return stringBuilder.toString();
    }
}

File: app/src/main/java/com/soullistener/viewtransfer/model/ViewTreeBean.kt
package com.soullistener.viewtransfer.model

import java.io.Serializable
import java.util.*

/**
 * @description
 * @author kuan
 * Created on 2018/12/3.
 */

data class ViewTreeBean(var childViews : ArrayList<ViewTreeBean>, val viewInfo : ViewSaveBean) : Serializable {

    /**存储子View */

    /**存储本View信息 */

}

/**
 * 保存View的数据信息
 */
data class ViewSaveBean(val viewContent: ViewContent, val viewResource: ViewResource, val viewInfo: ViewInfo) : Serializable {

    /**
     * 储存view的内容
     */
    data class ViewContent(val name: String = "", val content: String = ""): Serializable {
        override fun toString(): String {
            return super.toString() + "name:$name ,content:$content"
        }
    }

    /**
     * 储存view的背景等
     */
    data class ViewResource(val backColor: Int = 0,val backImage : String = "") : Serializable{
        override fun toString(): String {
            return super.toString() + "backColor:$backColor,backImage:$backImage"
        }
    }

    /**
     * 储存view的坐标、宽高
     */
    data class ViewInfo(val x: Int = 0, val y: Int = 0, val width: Int = 0, val height: Int = 0) : Serializable {
        override fun toString(): String {
            return super.toString() + "x:$x,y:$y,width:$width,height:$height"
        }
    }

}

/**
 * View列表的个数
 */
data class ViewNumber(val number: Int) : Serializable